### PR TITLE
Gracefully handle JSON.parse errors

### PR DESCRIPTION
**Pull request by AlexTugarev**
This PR improves handling of parse errors for incoming message.

### CODE REVIEW

File: packages/vscode-ws-jsonrpc/src/socket/reader.ts
Comment (CGNonofr) [inline packages/vscode-ws-jsonrpc/src/socket/reader.ts:70]: I have a few remarks:
- What about using the Error constructor?
- What does that `name` property means? and why 400 as a string?
- `typeof err === 'object'` doesn't mean it's an error, what about using `err instanceof Error`? (the cast wouldn't be required then

**Review by CGNonofr — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Were you experiencing issues or is this theoretical?

Or does this change comes from somewhere particular?

File: packages/vscode-ws-jsonrpc/src/socket/reader.ts
Comment (AlexTugarev) [reply packages/vscode-ws-jsonrpc/src/socket/reader.ts:70]: @CGNonofr, thanks for the questions. 'Trying to answer as much as possible. 

Here we're creating an error response, which is sent over the wire back to the client. This error is not to be further handled in this library. 
Also, this is a repetition of a pattern you can find above in this file: https://github.com/TypeFox/monaco-languageclient/blob/2087bd3ebe040ec9c5e986d7146625db64c607f0/packages/vscode-ws-jsonrpc/src/socket/reader.ts#L25

> What does that name property means? and why 400 as a string?

Error code `400` to signal a bad request. Again, following the example to show consistency, which should later on allow for improvement of similar cases. 

> doesn't mean it's an error

There isn't anything. Just trying to read the message from something that looks like an error, even if it wasn't constructed as an error. In any case, including 'unknown' is the fallback, and just in case we get to read a message, that will do.

**Comment by CGNonofr** [comment]
@kaisalmen I'll let you have a look and approve, LGTM

**Review by CGNonofr — APPROVED** [review]

**Comment by kaisalmen** [comment]
I have no objections. Who is going to push the button?

**Comment by kaisalmen** [comment]
@AlexTugarev I guess you require a new release?

**Comment by AlexTugarev** [comment]
👋🏻 @kaisalmen! 
Indeed, it would be awesome if`vscode-ws-jsonrpc` would be updated soon.

 🦊 🍻

**Comment by kaisalmen** [comment]
Here you go: https://www.npmjs.com/package/vscode-ws-jsonrpc/v/2.0.2

**Pull request merged**
